lib/bugsnag.rb

```diff
@@ -34,7 +34,9 @@ def configure
 
     # Explicitly notify of an exception
     def notify(exception, auto_notify=false, &block)
-      if auto_notify && !configuration.auto_notify
+      report = Report.new(exception, configuration, auto_notify)
```

Comment:
Why create the report before the validation blocks? This is extra processing when a user is running in a testing/debug mode.